geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/GeoMesaMetadata.scala

```diff
@@ -8,254 +8,280 @@
 
 package org.locationtech.geomesa.accumulo.data
 
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.TimeUnit
+
+import com.google.common.cache.{CacheBuilder, CacheLoader}
 import com.typesafe.scalalogging.LazyLogging
-import org.apache.accumulo.core.client.impl.{MasterClient, Tables}
-import org.apache.accumulo.core.client.mock.MockConnector
 import org.apache.accumulo.core.client.{Connector, Scanner}
 import org.apache.accumulo.core.data.{Mutation, Range, Value}
-import org.apache.accumulo.core.security.ColumnVisibility
-import org.apache.accumulo.core.security.thrift.TCredentials
-import org.apache.accumulo.trace.instrument.Tracer
 import org.apache.hadoop.io.Text
 import org.locationtech.geomesa.accumulo.AccumuloVersion
 import org.locationtech.geomesa.accumulo.data.AccumuloBackedMetadata._
-import org.locationtech.geomesa.accumulo.util.{EmptyScanner, GeoMesaBatchWriterConfig, SelfClosingIterator}
-import org.locationtech.geomesa.security.AuthorizationsProvider
+import org.locationtech.geomesa.accumulo.util.{EmptyScanner, GeoMesaBatchWriterConfig}
 
 import scala.collection.JavaConversions._
-import scala.collection.mutable
 
 /**
  * GeoMesa Metadata/Catalog abstraction using key/value String pairs storing
- * them on a per-featurename basis
+ * them on a per-typeName basis
  */
-trait GeoMesaMetadata {
-  def delete(featureName: String, numThreads: Int)
+trait GeoMesaMetadata[T] {
```

Comment:
Just curious, are there plans for the other data stores like the Kafka data store to support stats as well?